src/app/shared/expiration/date-picker/expiration-date.component.ts

```diff
@@ -78,15 +78,12 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
   }
 
   restoreDefaultHandler(): void {
+    this.defaultExpirationDate();
     this.add.emit(this.defaultValidityPeriod);
-    this.expirationDate.setValue(
-      new Date(Date.now() + this.defaultValidityPeriod)
-    );
-    this.expirationTimeShift = this.defaultValidityPeriod;
```

Comment:
This line doesn't seem to be in the new logic. Is this expected?

Reply:
It is, in method `setDefaultExpirationDate` (changed it from `defaultExpirationDate`).